### PR TITLE
feat: develop links & resources page

### PR DESCRIPTION
**Pull request by sakshi-choudhary**
**ISSUE #310**

**Describe the changes you've made**

Developed a Links & Resources page.

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**
NA

**Additional context (OPTIONAL)**
I have rendered conditionally so if the link / doc does not have a preview image, it will be rendered without an image.

**Checklist**

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**
Preview: [https://deploy-preview-311--edu-client.netlify.app/resources/:coursename](https://deploy-preview-311--edu-client.netlify.app/resources/:coursename)
 

<!-- Please add deployed link to here text >

**PREVIEW**

img

### CODE REVIEW

**Comment by sakshi-choudhary** [comment]
@Abhishek-kumar09 Please review!

**Comment by Abhishek-kumar09** [comment]
Give some more left and right margin.

**Review by Abhijay007 — COMMENTED** [review]
@sakshi-choudhary I reviewed your code and there are some minor changes that I want to suggest:

1. Instead of showing links and resources in a unordered format you can make a subsection for both like the upper one for "links" and the lower one for "resources" or downloadable material. means it looks more ordered and supervised to the user and provided a good user experience in my opinion. 

2. Avoid writing links as subtext like it looks good without them,  instead of showing links just show the name of the topic or Another thing you can do is to mention the site or source name as subtext.

these are just suggestions 😄. it is up to you if u want to implement these or not.

**Comment by sakshi-choudhary** [comment]
> @sakshi-choudhary I reviewed your code and there are some minor changes that I want to suggest:
> 
> 1. Instead of showing links and resources in a unordered format you can make a subsection for both like the upper one for "links" and the lower one for "resources" or downloadable material. means it looks more ordered and supervised to the user and provided a good user experience in my opinion.
> 2. Avoid writing links as subtext like it looks good without them,  instead of showing links just show the name of the topic or Another thing you can do is to mention the site or source name as subtext.
> 
> these are just suggestions 😄. it is up to you if u want to implement these or not.

I think this doesn't make sense in separating the links, resource is a resource and I kept  full link text as well so that incase the link is not opening by clicking user can copy the link and open it in another tab.

**Comment by sakshi-choudhary** [comment]
@Abhishek-kumar09 Please check now, I've given margin in -axis

**Comment by Abhijay007** [comment]
> I kept full link text as well so that incase the link is not opening by clicking user can copy the link and open it in another tab.

It is nice as you provide a "link address" to the user to use it in case the link is not working, but it is hard to copy that link because there is not an option to copy that link, and in mobile devices (as far as I know) long presses to copy text is not supported in Apps. You can add a option to copy the text on click or provide a clipboard icon with the same functionality.

**Comment by sakshi-choudhary** [comment]
> > I kept full link text as well so that incase the link is not opening by clicking user can copy the link and open it in another tab.
> 
> It is nice as you provide a "link address" to the user to use it in case the link is not working, but it is hard to copy that link because there is not an option to copy that link, and in mobile devices (as far as I know) long presses to copy text is not supported in Apps. You can add a option to copy the text on click or provide a clipboard icon with the same functionality.

Yes, this will be convenient for users, I'll implement. Thanks :)

**Comment by sakshi-choudhary** [comment]
@Abhishek-kumar09 I have implemented  `copy link to clipboard` feature as well. Please do a review!

File: src/pages/Resources /index.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+          anchorOrigin={{
+            vertical: "bottom",
+            horizontal: "left",
+          }}
+          open={open}
+          autoHideDuration={6000}
+          onClose={handleClose}
+          message=" Copied to Clipboard!"
+          action={
+            <React.Fragment>
```
Comment (Abhishek-kumar09) [inline src/pages/Resources /index.js:164]: replace it with `<>`

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

**Comment by sakshi-choudhary** [comment]
@Abhishek-kumar09 ping!

**Review by Abhishek-kumar09 — APPROVED** [review]

**Pull request merged**